libdebug/commlink/pipe_manager.py

```diff
@@ -381,75 +502,118 @@ def sendlineafter(
-                        if recv_stdout == b"\n":
-                            break
-                except RuntimeError:
-                    # The child process has closed the stdout pipe
-                    liblog.warning("The stdout pipe of the child process is not available anymore")
-                    stdout_is_open = False
-                    continue
-            if stderr_is_open:
-                try:
-                    while recv_stderr := self._recv(numb=1, timeout=0.05, stderr=True):
-                        sys.stderr.write(payload=recv_stderr)
-                        if recv_stderr == b"\n":
-                            break
-                except RuntimeError:
-                    # The child process has closed the stderr pipe
-                    liblog.warning("The stderr pipe of the child process is not available anymore")
-                    stderr_is_open = False
-                    continue
-
-    def interactive(self: PipeManager, prompt: str = prompt_default) -> None:
+            # To avoid starvation, we switch between pipes upon receiving a bunch of data from one of them.
+            if self._stdout_is_open:
+                while True:
+                    new_recv = self._raw_recv()
+                    payload = self.__stdout_buffer.get_data()
+
+                    if not (new_recv or payload):
+                        # No more data available in the stdout pipe at the moment
+                        break
+
+                    sys.stdout.write(payload)
+                    self.__stdout_buffer.clear()
+            elif not stdout_has_warned:
+                # The child process has closed the stdout pipe and we have to print the warning message
+                liblog.warning("The stdout pipe of the child process is not available anymore")
+                stdout_has_warned = True
+            if self._stderr_is_open:
+                while True:
+                    new_recv = self._raw_recv(stderr=True)
+                    payload = self.__stderr_buffer.get_data()
+
+                    if not (new_recv or payload):
+                        # No more data available in the stderr pipe at the moment
+                        break
+
+                    sys.stderr.write(payload)
+                    self.__stderr_buffer.clear()
+            elif not stderr_has_warned:
+                # The child process has closed the stderr pipe
+                liblog.warning("The stderr pipe of the child process is not available anymore")
+                stderr_has_warned = True
+
+    def interactive(self: PipeManager, prompt: str = prompt_default, auto_quit: bool = False) -> None:
         """Manually interact with the child process.
 
         Args:
             prompt (str, optional): prompt for the interactive mode. Defaults to "$ " (prompt_default).
+            auto_quit (bool, optional): if automatically quit the interactive mode when the child process is not running. Defaults to False.
```

Comment:
*whether to* automatically quit?

Reply:
it's longer

Reply:
Yes, but "if automatically quit" is not grammatically correct